README.md

```diff
@@ -1 +1,136 @@
+  - [🤝 Contributing ](#-contributing-)
+  - [⭐️ Show your support ](#️-show-your-support-)
+  - [🙏 Acknowledgments ](#-acknowledgments-)
+  - [📝 License ](#-license-)
+
+# 📖 TDD Project <a name="about-project"></a>
+
+**TDD-Project** This repository contains a Ruby project demonstrating Test-Driven Development (TDD) principles. The project includes a Solver class with methods for factorial calculation, word reversal, and fizzbuzz generation. The implementation is accompanied by unit tests using RSpec.
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+- Ruby
+  <details>
+    <summary>Language</summary>
+    <ul>
+      <li><a href="https://www.ruby-lang.org/en/">Ruby</a></li>
+    </ul>
+  </details>
+
+### Key Features <a name="key-features"></a>
+
+- **[test driven development]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 💻 Getting Started with tdd-project <a name="getting-started"></a>
+
+### Prerequisites
+
+In order to run this project you need:
+
+1. Ruby installed on your machine
+2. Text editor (VSCode, Sublime, Atom, etc.)
+
+### Setup
+
+Clone this [repository](https://github.com/ernestkorir/TDD-project.git) to your desired folder:
+
+```sh
+  cd my-folder
+  git clone https://github.com/ernestkorir/TDD-project.git
+```
+
+### Install
+
+Install this project with:
+
+```sh
+  cd TDD-project
+```
+
+### Usage
+
+Run this project with:
+
+```sh
+  rspec
+```
+
+### Run tests
+
+Run tests with:
+
+```sh
+  rubocop
+```
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 👥 Authors <a name="authors"></a>
+
+👤 **Tom Daniel**
+
+- GitHub: [@tomdan-ai](https://github.com/tomdan-ai)
+- Twitter: [@kvngWebstar](https://twitter.com/tomudoh1)
+- LinkedIn : [Tom Udoh](https://linkedin.com/in/tomudoh)
+
+👤 **Ernest Korir**
+
+- GitHub: [@ernestkorir](https://github.com/ernestkorir)
+- LinkedIn: [Ernest korir](https://www.linkedin.com/in/ernest-korir/)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🔭 Future Features <a name="future-features"></a>
+
+- Take in a string and return a number
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/ernestkorir/TDD-project/issues).
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project, Give a ⭐️.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- Project from [Microverse](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md) Ruby module
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT) licensed.
```

Comment:
- [ ] The link to the licence is not working as expected because you're referring to MIT but the file is called MIT.md.